### DISCUSSION TITLE
[Featurerequest] ESP12F_Relay_X8 - increase number of timers

### DISCUSSION THREAD

**Discussion by Cysign** [arendst/Tasmota#22306] [Ideas]
Hi, I just flashed a ESP12F_Relay_X8 using tasmota.
I'd like to replace multiple Sonoff S20 single wifi sockets with the relay-board.

Sonce I'm using the S20 wifi sockets for warterin my smartgardening-system, I would need multiple timers on every output.

At lease for the waterpumps (I need two different) I need up to 4 timers per day.
For the fan (many plants need artificial wind when growing indoor to gain more strength) I usually have a fan powered 2x 30min per day.

It looks like I would have 16 'events' on tasmota.

So 2 waterpumps (which equaly two relays) * 4 events/day * two actions (each one and off seems to be an action) makes 16 timers. So I cannot use the other 6 relays.

Any Idea for a workaround to get more timers/events per day?
Or any possibility to set starttime+duration per timer? So I wouldn't need two timers per event (on + off).

For a single- or double-relay device 16 timers should be fair enough. But for 8 relays that's toop less...

**Comment by sfromis** [comment]
If you use the 16 timers as rule triggers for a time-of-day, you could then have rules doing combined things, like having the duration to power off again, and possibly if you want more than one pump/fan at the same time.

But if you want a lot better options for local control, you could use a board based on ESP32 instead of ESP8266, then you'd have the programming language Berry, providing extensive possibilities for local programming.

**Comment by Cysign** [comment]
So for rules I first need to define them in console and then apply it to the timer?
Maybe that would do the trick in my case. Thanks!

For the timer: Does the module time-sync on every boot, once I set timezone?

**Reply by sfromis** [reply]
If what you mean is getting correct time, Tasmota will synchronize after boot and hourly, by default with public NTP servers, also with default time zone setting. Using https://tasmotatimezone.com/ can be recommended for setup here.

**Comment by Cysign** [comment]
Just for my understanding:

Relay 1 Light (either 18h or 12h is needed).
Relay 2 fan.
Relay 3 waterpump.

Let's say I'd like to turn on relay 1+2+3 at 9am.
Relay 1 needs to be on for 18h.
Relay 2 needs to be on for 30 minutes on 9am and 9pm.
Relay 3 needs to be on 4 times a day for 65 seconds.
So what would my rolus look like?

And what would I have to do to switch from 65seconds watering time to 80 seconds?
Would that be a modification of the rules or would I need a new rule? So I would have to choose the 65seconds watering-rule for the first 3 weeks and after that I would need another rule for watering 80 seconds?

Btw. I have a sonoff basic here with updated tasmota, so I can test rules without changing something on my 8-relay-board if required.

**Reply by sfromis** [reply]
Your rules would depend a lot on what timers you choose to use, and how, like if you want each change in something to triggered by a timer. Then each rule could be `on Clock#timer==3 do Power1 off endon` etc.

For intervals not whole minutes, it would make sense to use TimedPower command, like `TimedPower3 65000,on` for 65 seconds on before off again.

Likewise with 30-minutes-times for relay 2. You could even do that for Relay 1 too. What works best for you depends on whether you want to control it all in the timers, or you are ok with modifying both timers and rules for different schedules.

Instead of using the timers feature at all, you could also have rules like `on Time#minute=480 do ... endon` where the number is minutes since midnight. That way you could have the whole schedule as a long list of rules.
